std/utf.d

```diff
@@ -1164,8 +1164,10 @@ package template codeUnitLimit(S)
  * Returns:
  *      decoded character
  */
-private dchar decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar = UseReplacementDchar.no, S)(auto ref S str, ref size_t index)
-    if (is(S : const char[]) || (isInputRange!S && is(Unqual!(ElementEncodingType!S) == char)))
+private RV decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar, RV = dchar, S)
```

Comment:
looks better, but I just hate auto as return type. Especially, as dchar is not that hard to understand.